### PR TITLE
ft: blacklist object prefixes

### PR DESCRIPTION
**Pull request by electrachong**
Dependency of https://github.com/scality/Mystique/pull/40

Add blacklisting for object prefixes.

### CODE REVIEW

**Comment by electrachong** [comment]
@ironman-machine try

File: lib/s3routes/routesUtils.js [hunk 1]
```diff
@@ -782,6 +782,28 @@ const routesUtils = {
...
     /**
+     * Validate object key per naming rules and restrictions
+     * @param {string} objectKey - object key
+     * @param {string[]} prefixBlacklist - prefixes reserved for internal use
+     * @return {object} - object containing true/false result and invalidPrefix
+     *  if false
+     */
+    isValidObjectKey(objectKey, prefixBlacklist) {
+        let invalidPrefix;
+        prefixBlacklist.some(prefix => {
```
Comment (jonathan-gramain) [inline lib/s3routes/routesUtils.js]: If you don't use the return value of `some` you don't need to return any value from the test function, and then there's no benefit of using `some`. A more concise way can be to use the `Array.find` function like `invalidPrefix = prefixBlacklist.find(prefix => objectKey.startsWith(prefix))`

Comment (electrachong) [reply lib/s3routes/routesUtils.js]: you're right, thanks for the tip!

**Comment by LaurenSpiegel** [comment]
@ironman-machine r+

**Pull request merged**